modal/app.py

```diff
@@ -879,15 +879,17 @@ def cls(
         if _warn_parentheses_missing:
             raise InvalidError("Did you forget parentheses? Suggestion: `@app.cls()`.")
 
+        # Argument validation
         if interactive:
             deprecation_error(
                 (2024, 5, 1), "interactive=True has been deprecated. Set MODAL_INTERACTIVE_FUNCTIONS=1 instead."
             )
 
-        if image is None:
-            image = self._get_default_image()
-
-        secrets = [*self._secrets, *secrets]
+        scheduler_placement = _experimental_scheduler_placement
+        if region:
```

Comment:
It would probably make sense to move out argument validation/auto-fixing into its own helper function(s) that can be used by both function, cls and Sandbox.create, but lets do that separately :)